### DISCUSSION TITLE
Request for a step-by-step guide on using 'ink' or pen.

### DISCUSSION THREAD

**Discussion by mhmd-cli** [sumatrapdfreader/sumatrapdf#4741] [Q&A]
I was looking through the documentation in the annotations part, but couldn't find any info on this. If anyone knows how to make ink annotations, please so share.

**Comment by GitHubRulesOK** [comment]
You should note that ink is NYI (not yet implemented) and a known shortage covered by issue https://github.com/sumatrapdfreader/sumatrapdf/issues/2485
SumatraPDF can share a file under 32MB with MS Edge and have shown how to save Edge inking whilst viewing with SumatraPDF.

Inking is also available in core MuPDF-GL but not so easy to use there.
![image](https://github.com/user-attachments/assets/730b5fbf-1366-4a34-9344-954561d4433b)

**Reply by mhmd-cli** [reply]
Okay, thanks.

**Reply by GitHubRulesOK** [reply]
if a file under 32MB is open in both (without any other file locking) then a save in edge will appear in SumatraPDF and also possibly the other way round with a reload in Edge.
Thus I have a shortcut to open current file/page in Edge for Read aloud, Inking or any other it may do "better"

**Reply by mhmd-cli** [reply]
I mean sound's limiting for a file that's just 32mb, is there an easy fix to this? Like if I got a lot of images in my pdf or epub whatever. That would be greatly appreciated if you can share that.

**Reply by GitHubRulesOK** [reply]
Hmm the older limit was 10 MB
It was a legacy and is current feature for using SumatraPDF with LaTeX Academic Article / Thesis editors where files are usually small.

There was much older multiformat experimental annotation system for only highlights (rectangles), but dropped after 3.2.
Dual working cannot generally work with any format apart from PDF. (PDF LaTeX being the reason for implementation)

a dicussion on abilities was in https://github.com/sumatrapdfreader/sumatrapdf/discussions/2940

**Reply by mhmd-cli** [reply]
Personally, I'm a vim user, and what really attracted me to Sumatura PDF was its vast use of key binds on Windows over several of the other available options with similar capabilities. I also read a lot of academic papers/articles, but I was using Xournal++ before this for that (for taking notes, highlights, annotations, etc). So, with the pen missing on those, it can't really be a universal solution for me.

Thanks for the help regardless, you've provided me with a lot of useful information. And it's not like I'm trying to undermine SumaturaPDF, the app does do what it claims to at the end of the day.

**Reply by GitHubRulesOK** [reply]
Yes one key press can call a batch file to open the same page in a single MSEdge window where a save will return the contents to SumatraPDF and the single edge window be closed.
So is 2 windows from one key press, a wiggle and a click (or Alt F4) and perhaps one from VIM opening SumatraPDF

**Reply by GitHubRulesOK** [reply]
I feel a newer batch file example may be needed let me try.

**Comment by GitHubRulesOK** [comment]
@mhmd-cli 

NOTE needs the latest pre-release version 

One key to open a window for inking. **Due to the way MS opens a single Edge window on the left** use SumatraPDF in right half of screen ! 
To be able to add inking into a SumatraPDF page I have written a script that will open a MS Edge window for inking PDF. Once done **use the save icon or CTRL+S** will show the ink in SumatraPDF, when finished after last save close the window.

There are 2 possible problems 
1) I had problems opening a file with odd characters in the filename
2) it may not always work with PDF files larger than 32 MB so use a copy of a file for testing.

![image](https://github.com/user-attachments/assets/ec99c88a-2824-4d2d-8a31-f2b4de788060)
![image](https://github.com/user-attachments/assets/60cebe2e-3538-4ac8-aab3-7eba8d055481)

The script 
---
With instructions inside it. I call mine DualEdit.cmd and use a "Plus" folder in SumatraPDF users local \SumatraPDF folder
```
@echo off
goto main
The aim of this file is to allow for a page in SumatraPDF to be passed to an
external viewer editor similar to the way LaTeX is used for edit and view.

For example if your testing a basic text based PDF in MS notepad or using
inking in Edge the updates there (when saved) will apear in SumatraPDF.

This is an early draft mainly for proof of concept, notepad and wordpad are not
functional, you can add browser replacements i.e. Chrome or Firefox (has inking).

In SumatraPDF Advanced Options file add to the external viewers section

ExternalViewers [
	[
		CommandLine = "C:\Users\K\AppData\Local\SumatraPDF\plus\DualEdit.cmd" edge reserved %p "%1"
		Name = Edge-it
		Filter = *.pdf
		Key = w
	]
]

I recommend a subdirectory called plus in the same folder as you may want to add more functions


:help
echo Usage: [ Edge ^| Notepad ^| Wordpad ] RESERVED 20 C:\test\file.pdf
echo We expect 1st agument is an app name such as Notepad or Edge to allow for multiple usage.
echo 2nd we expect RESERVED for future use, 3rd just the page # and 4th the full/path/filename.
echo/
echo Side note page number 20 could be also 0x14 (Hex) or 10+10 BEWARE 024 is Octal for 20 !
echo/
pause
exit /b

:main
REM echo %1 %2 %3 %4 & pause

REM test the 4th is a valid file
if not exist "%~4" goto help
REM test for now 2nd is RESERVED or reserved
if /i not %2 == RESERVED goto help
REM test 3rd is only a number
set "pagenum=%3"
set /a "test=0+%pagenum%"
if %test% == 0 goto help

for %%f in (edge notepad wordpad) do if /i %%f == %1 goto %%f
exit /b

:edge
"%ProgramFiles%\Microsoft\Edge\Application\msedge.exe" --profile-directory=Default --app="file:///%~dpnx4#page=%test%"
exit /b

:notepad
rem needs a vbs script used for goto page
 exit /b

:wordpad
rem needs a vbs script used for goto page
exit/b
```
